src/cljc/athens/common_db.cljc

```diff
@@ -275,152 +280,151 @@
   (->> (d/pull db '[* :block/_refs] [:node/title page-title])
        :block/_refs
        (mapv :db/id)
-       (merge-parents-and-block db)
-       group-by-parent
-       (sort-by :db/id)
-       vec
-       rseq))
+       (mapv #(d/pull db '[:db/id :node/title :block/uid :block/string] %))))
 
 
 (defn- extract-tag-values
-  "Extracts `tag` values with `extractor-fn` from parser AST."
-  [ast tag extractor-fn]
-  (->> (tree-seq vector? extractor-fn ast)
+  "Extracts `tag` values from `children-fn` children with `extractor-fn` from parser AST."
+  [ast tag-selector children-fn extractor-fn]
+  (->> (tree-seq vector? children-fn ast)
        (filter vector?)
-       (keep #(when (= tag (first %))
+       (keep #(when (tag-selector (first %))
                 (extractor-fn %)))
        set))
 
 
-(defn- extract-page-links
-  "Extracts from parser AST `:page-link`s"
-  [ast]
-  (extract-tag-values ast :page-link #(cond
-                                        (and (vector? %)
-                                             (< 2 (count %)))
-                                        (nth % 2)
-                                        (and (vector? %)
-                                             (< 1 (count %)))
-                                        (nth % 1)
-                                        :else
-                                        %)))
-
+(defn strip-markup
+  "Remove `start` and `end` from s if present.
+   Returns nil if markup was not present."
+  [s start end]
+  (when (and (string/starts-with? s start)
+             (string/ends-with? s end))
+    (subs s (count start) (- (count s) (count end)))))
+
+
+(defn string->lookup-refs
+  "Given string s, compute the set of refs expressed as Datalog lookup refs."
+  [s]
+  (let [ast (parser/parse-to-ast s)
+        block-ref-str->uid #(strip-markup % "((" "))")
+        page-ref-str->title #(or (strip-markup % "#[[" "]]")
+                                 (strip-markup % "[[" "]]")
+                                 (strip-markup % "#" ""))
+        block-lookups (into #{}
+                            (map (fn [uid] [:block/uid uid]))
+                            (extract-tag-values ast #{:block-ref} identity #(-> % second :from block-ref-str->uid)))
```

Comment:
From a performance point of view probably not, since that's more work for the parser and the parser is used elsewhere.

I think it might be worth it if there's more places other than linkmaker that need that, or other similar cases of "parser needs to give more info". But for now this seems ok.